src/asmjit/core/virtmem.cpp

```diff
@@ -250,10 +255,37 @@ static Error asmjitErrorFromErrno(int e) noexcept {
   }
 }
 
+static uint32_t getLargePageSize() noexcept {
+#if defined(__APPLE__) && defined(VM_FLAGS_SUPERPAGE_SIZE_2MB) && ASMJIT_ARCH_X86
+  return 2*1024*1024;
+#elif defined(__FreeBSD__)
+  Support::Array<size_t, 2> pageSize;
+  return (getpagesizes(pageSize.data(), 2) < 2) ? 0 : uint32_t(pageSize[1]);
+#elif defined(__linux__)
+  uint32_t largePageSize = 0;
+  int fd = ::open("/sys/kernel/mm/transparent_hugepage/hpage_pmd_size", O_RDONLY);
+  if (fd != -1) {
+    Support::Array<uint8_t, 11> buffer; // enough for `ceil(log10(pow(2, 32))) + 1`
+    ssize_t length = ::read(fd, buffer.data(), buffer.size());
+    ::close(fd);
+    // Try to parse the contents of `hpage_pmd_size` assuming the input has up to 10 decimal digits and is terminated
+    // with a newline character.
+    ASMJIT_ASSERT(length > 0);
+    ASMJIT_ASSERT(buffer[length - 1] == '\n');
```

Comment:
I would turn this into a condition instead - I the kernel returns some nonsense let's not do something weird.

Reply:
Sure, I'll turn this into an `if` that guards the loop.